pkg/kubectl/cmd/logs.go

```diff
@@ -199,7 +225,40 @@ func (o LogsOptions) Validate() error {
 
 // RunLogs retrieves a pod log
 func (o LogsOptions) RunLogs() (int64, error) {
-	req, err := o.LogsForObject(o.Object, o.Options)
+	switch t := o.Object.(type) {
```

Comment:
My proposed changes are I've included in https://github.com/fraenkel/kubernetes/pull/1